### PR TITLE
Feature/#739 vault search ui

### LINKED ISSUE #739: Replay search is too complicated
There are lots of essentially useless search terms and too many dropdown menus and items.

The python client search is much better, you have a field for player name, a field for map name, a dropdown to select the base mod(1v1ladder, faf, fafbeta, phantom etc.) and a field for minimum rating.

### LINKED ISSUE #1231: Rework vault, user friendlyness

### PR DESCRIPTION
**Pull request by Sheikah45**
This will implement the filters for the vault search as well as make the replays more visible.
It utilizes controls from controlsfx.

Fixes #739 and related

- [x]  Finish adding filter types
- [x]  Add to Map and Mod Vault
- [x]  Add ability to save queries and run again
- [x]  Move Replay Tab
- [x]  Add indication filter is used
- [x]  Add unit tests

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#1944: better utilities for vault  searches (custom search presets & slider for min max rating) ](https://issuehunt.io/repos/32919066/issues/1944)
---
</details>

### CODE REVIEW

**Comment by Sheikah45** [comment]
Current UI
Replays 
Screenshot (155)


Maps
Screenshot (156)


Mods
Screenshot (157)


Saved Queries
Screenshot (141)

Date Tool
Screenshot (147)

Range Tool
Screenshot (148)

Combo Check Tool
Screenshot (149)

**Comment by Brutus5000** [comment]
We could simplify ranked map to "[x] only ranked" checkbox and validity "[x] only rated" checkbox.
I doubt any regular user will search for replays of a certain invalidity reason.

**Comment by Brutus5000** [comment]
The rating needs a tiny rework for future compatibility. My suggestion: select a leaderboard and then use a from/to.

Furthermore thoughts on the controls:
The selectable combobox of course for dropdown of e.g. multiple game modes.
And a range slider for the rating from 0 to 3000

Both exist in [controls-fx](https://github.com/controlsfx/controlsfx/wiki/ControlsFX-Features)

**Comment by Sheikah45** [comment]
> Both exist in [controls-fx](https://github.com/controlsfx/controlsfx/wiki/ControlsFX-Features)

Yes I am currently using those I should add an update to the pictures

**Comment by Sheikah45** [comment]
> The rating needs a tiny rework for future compatibility. My suggestion: select a leaderboard and then use a from/to.

Is the api for this already implemented in the client do you know? If not I am not sure I would want to introduce it in this pr specifically

**Comment by Sheikah45** [comment]
> 
> 
> We could simplify ranked map to "[x] only ranked" checkbox and validity "[x] only rated" checkbox.
> I doubt any regular user will search for replays of a certain invalidity reason.

I can simplfy it but might need two check boxes as people may want to return only not ranked replays maybe?

**Comment by Brutus5000** [comment]
> I can simplfy it but might need two check boxes as people may want to return only not ranked replays maybe?

The "people may want to search for <anything>" is exactly what got critized as it makes the whole thing complicated. I don't think there is non-power-user use case for "search particularly unrated games". It's either all or just the rated games (as they tend to be more serious, better build order etc)

> Is the api for this already implemented in the client do you know? If not I am not sure I would want to introduce it in this pr specifically

No, otherwise the other leaderboard issue would be solved already.

**Comment by Sheikah45** [comment]
Yeah I wasnt sure if only having a single check box would confuse from a ux perspective in terms of if it means I am getting not ranked or all but might not be a problem

**Comment by Sheikah45** [comment]
Alright this should be all set now

**Comment by Sheikah45** [comment]
alright this should be all set now

File: src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java [hunk 1]
```diff
@@ -110,6 +115,87 @@ protected Node getDetailView() {
     return replayDetailController.getRoot();
   }
 
+  protected void initSearchController() {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java:114]: Method too long

Comment (Sheikah45) [reply src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java:114]: Broke it up

**Comment by Sheikah45** [comment]
Hoping to merge this soon

**Pull request merged**